sfa_dash/blueprints/data_listing.py

```diff
@@ -33,60 +36,87 @@ def __init__(self, data_type, **kwargs):
-        breadcrumb_format = '/<a href="{url}">{text}</a>'
-        breadcrumb = ''
-        if self.data_type == 'cdf_forecast_group':
-            type_label = 'CDF Forecast'
+    def get_breadcrumb_dict(self, **kwargs):
+        """Build the breadcrumb dictionary for the listing page. If site_id
+        or aggregate_id are passed as key word arguments, prepends the
+        appropriate links. Note that these links depend on the existence of
+        the self.location_data attribute, that should only be set by during
+        a successful fetch of the site/aggregates metadata.
+        """
+        breadcrumb_dict = OrderedDict()
+        human_label = human_friendly_datatype(self.data_type)
+        if 'site_id' in kwargs:
+            breadcrumb_dict['Sites'] = url_for('data_dashboard.sites')
+            breadcrumb_dict[self.location_data['name']] = url_for(
+                'data_dashboard.site_view', uuid=kwargs['site_id'])
+        elif 'aggregate_id' in kwargs:
+            breadcrumb_dict['Aggregates'] = url_for(
+                'data_dashboard.aggregates')
+            breadcrumb_dict[self.location_data['name']] = url_for(
+                'data_dashboard.aggregate_view', uuid=kwargs['aggregate_id'])
+        breadcrumb_dict[f'{human_label}s'] = url_for(
+            f'data_dashboard.{self.data_type}s', **kwargs)
+        return breadcrumb_dict
+
+    def get_subnav_kwargs(self, **kwargs):
+        """Creates a dict to be used when formating the sub navigation. The
+        resulting dict is used to format the fstring keys found in the
+        DataListingView.subnav_format variable.
+        """
+        subnav_kwargs = {}
+        if 'aggregate_id' in kwargs:
+            subnav_kwargs['observations_url'] = url_for(
+                'data_dashboard.aggregate_view', uuid=kwargs['aggregate_id'])
         else:
-            type_label = self.data_type.title()
-        if site_id is not None:
-            site_metadata = handle_response(
-                sites.get_metadata(site_id))
-            breadcrumb += breadcrumb_format.format(
-                url=url_for('data_dashboard.sites'),
-                text='Sites')
-            breadcrumb += breadcrumb_format.format(
-                url=url_for('data_dashboard.site_view', uuid=site_id),
-                text=site_metadata['name'])
-        breadcrumb += breadcrumb_format.format(
-            url=url_for(f'data_dashboard.{self.data_type}s', uuid=site_id),
-            text=type_label)
-        return breadcrumb
+            subnav_kwargs['observations_url'] = url_for(
+                'data_dashboard.observations', **kwargs)
+        subnav_kwargs['forecasts_url'] = url_for(
+            'data_dashboard.forecasts', **kwargs)
+        subnav_kwargs['cdf_forecasts_url'] = url_for(
+            'data_dashboard.cdf_forecast_groups', **kwargs)
+        return subnav_kwargs
 
     def get_template_args(self, **kwargs):
```

Comment:
so kwargs is only ever aggregate_id or site_id right? I think it's more clear if arguments are explicit ``get_template_args(self, site_id=None, aggregate_id=None)`` so the rest of the function can use these. As far as I can tell this class isn't subclassed anywhere where this might be a problem, right?